From 9d016ace855b88916746837b1a424cf9e5960c3b Mon Sep 17 00:00:00 2001
From: Dexter Gaon-Shatford <dexter@gaonshatford.ca>
Date: Thu, 20 Oct 2022 20:06:16 -0400
Subject: [PATCH] Fix wrong pitch argument in call to SDL_CreateRGBSurfaceFrom.

This patch makes the change in
https://github.com/mupen64plus/mupen64plus-core/pull/970 compatible with
the source distribution of mupen64plus v2.5.9.

This patch should be removed when the next version of mupen64plus is
released since it will already contain the changes from the PR mentioned
above.
---
 src/api/vidext_sdl2_compat.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/source/mupen64plus-core/src/api/vidext_sdl2_compat.h b/source/mupen64plus-core/src/api/vidext_sdl2_compat.h
index 06d778dd..7cbd5963 100644
--- a/source/mupen64plus-core/src/api/vidext_sdl2_compat.h
+++ b/source/mupen64plus-core/src/api/vidext_sdl2_compat.h
@@ -474,8 +474,9 @@ SDL_SetVideoMode(int width, int height, int bpp, Uint32 flags)
         if (SDL_GL_MakeCurrent(SDL_VideoWindow, SDL_VideoContext) < 0) {
             return NULL;
         }
+        int pitch = (bpp + 7) / 8 * width;
         SDL_VideoSurface =
-            SDL_CreateRGBSurfaceFrom(NULL, width, height, bpp, 0, 0, 0, 0, 0);
+            SDL_CreateRGBSurfaceFrom(NULL, width, height, bpp, pitch, 0, 0, 0, 0);
         if (!SDL_VideoSurface) {
             return NULL;
         }
-- 
2.38.1

